platform/mds/mds-test-bundle/src/test/java/org/motechproject/mds/test/osgi/MdsDdeBundleIT.java

```diff
@@ -1984,6 +1994,18 @@ public void shouldCreateHistoryForManyToMany() {
         assertEquals(jackDt2, safeGetProperty(secondVersion, "modificationDate"));
     }
 
+    @Test
+    public void shouldRevertFromTrash() {
+        RevertFromTrash testRevert = testRevertFromTrashService.create(new RevertFromTrash("revertTest"));
```

Comment:
this test should look differently, try to test the real use case. First of all after refactor trash and history to use instance lifecycle listeners, an instance should be moved to trash while deleting instance. So do not use method moveToTrash, just delete the instance.

you always use id of created instance, in this example there are only one instance in main entity and one in trash, so the id probably is the same and has value 1. I think this test should be a bit more complicated. Can you create two instances,  delete them, then revert one of them from trash and check at the end if there is one instance in main table and one in trash